### PR TITLE
feat: add prettier config and test npm script

### PR DESCRIPTION
**Pull request by karstens**
## Description

* adding a prettier config to match our other repos/tools
* check all scripts / tasks to run via moon
* replace `yarn` with `pnpm` everywhere
* fix lint-stage and git hooks

## Motivation

* make code better

## To-do list

- [X] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).

- [x] This pull request is ready to merge.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,14 +7,19 @@
     "moon": "moon",
     "pre-commit": "lint-staged",
     "prepare": "husky install",
+    "prettier": "npm run prettier:code &&npm run prettier:ui",
+    "prettier:ui": "npx prettier --check --config .prettierrc \"docs/ui/src/**/*.*\"",
+    "prettier:code": "npx prettier --check --config .prettierrc \"packages/contrast-colors/lib/**/*.*\"",
+    "prettier:fix": "npm run prettier:code -- --write && npm run prettier:ui -- --write",
```
Comment (GarthDB) [inline package.json:13]: I think these are unnecessary. We should use prettier on everything that isn't excluded by the `.prettierignore` and defined in the [.lintstagedrc.json](https://github.com/adobe/leonardo/blob/monorepo/.lintstagedrc.json) file.

**Review by GarthDB — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (karstens) [reply package.json:13]: @GarthDB Thank you!! yes, I saw the `.lintstagedrc.json` entry, but I would like to be able to run `prettier` without doing a commit. Sorry, I am new to monorepo and not sure, if/where tools like prettier would be added to the monorepo? I saw this: https://moonrepo.dev/docs/guides/examples/prettier#setup
I will remove the npm script entries, but shouldn't I add a `.moon/tasks/node.yml`?

Comment (GarthDB) [reply package.json:13]: A node.yaml tasks file for type checking? Or are there other project wide node scripts you think should be in there?

Comment (karstens) [reply package.json:13]: @GarthDB Well, after our Slack chat, I think, it's okay to configure the IDE to run prettier on "file save". So, no need for that node.yaml at this point. But still curios, if that  `.moon/tasks/node.yml` is a good pattern for moon mono repos...

**Comment by karstens** [comment]
@GarthDB This is ready to merge.

I pushed two commits:
* one that replaced all npx with pnpm calls, so the tasks can be run via `moon` 
* one that fixed the husky/commit-msg, which was not set executable. 

Thanks

**Review by GarthDB — APPROVED** [review]

**Comment by GarthDB** [comment]
## Run report for [30ab9ac4](https://github.com/adobe/leonardo/commit/30ab9ac473ae8cbc5bae4afec2367d40a5799613)
Total time: 35.5s | Comparison time: 35.5s | Estimated savings: 46.4ms (0.1% faster)
|     | Action | Time | Status | Info |
| :-: | :----- | ---: | :----- | :--- |
| 🟩 | `SyncWorkspace` | 0ms | Passed |  |
| ⬛️ | `SetupNodeTool(^20.0.0)` | 1s | Skipped |  |
| 🟩 | `InstallNodeDeps(^20.0.0)` | 8.5s | Passed |  |
| 🟩 | `SyncNodeProject(contrast-colors)` | 0.2ms | Passed |  |
| 🟩 | `SyncNodeProject(ui)` | 0.2ms | Passed |  |
| 🟩 | `RunTask(contrast-colors:test)` | 4.7s | Passed |  |
| 🟩 | `RunTask(ui:buildSite)` | 12.6s | Passed |  |
| 🟩 | `RunTask(ui:clean)` | 800.9ms | Passed |  |
| 🟩 | `RunTask(ui:copyCNAME)` | 759.9ms | Passed |  |
| 🟩 | `RunTask(ui:copyIcons)` | 1.7s | Passed |  |
| 🟩 | `RunTask(ui:copyUIIcons)` | 641.2ms | Passed |  |
| 🟩 | `RunTask(ui:copyWorkflowIcons)` | 693.3ms | Passed |  |
| 🟥 | `RunTask(ui:deploySite)` | 8.6s | Failed |  |

<details><summary><strong>Touched files</strong></summary><div>

```
.github/workflows/ci.yml
.gitignore
.gitmodules
.husky/.gitignore
.moon/toolchain.yml
.moon/workspace.yml
.prettierignore
.prettierrc
README.md
docs/ui/README.md
docs/ui/moon.yml
docs/ui/package.json
docs/ui/src/APCAlookup.js
docs/ui/src/demo.css
docs/ui/src/demo.js
docs/ui/src/htmlHead.html
docs/ui/src/index.js
docs/ui/src/js/addBulkDialog.js
docs/ui/src/js/addColorsFromImage.js
docs/ui/src/js/addFromURL.js
docs/ui/src/js/addLightnessBulkDialog.js
docs/ui/src/js/addScaleBulkDialog.js
docs/ui/src/js/bulkConvertDialog.js
docs/ui/src/js/chroma-plus.js
docs/ui/src/js/colorDetailsPanel.js
docs/ui/src/js/colorScale.js
docs/ui/src/js/colorScaleDiverging.js
docs/ui/src/js/colorScaleQualitative.js
docs/ui/src/js/colorScaleSequential.js
docs/ui/src/js/colorWheel.js
docs/ui/src/js/compareColors.js
docs/ui/src/js/convertColors.js
docs/ui/src/js/create3dModel.js
docs/ui/src/js/createBaseScaleOptions.js
docs/ui/src/js/createChart.js
docs/ui/src/js/createDemos.js
docs/ui/src/js/createHtmlElement.js
docs/ui/src/js/createInterpolationCharts.js
docs/ui/src/js/createOutputColors.js
docs/ui/src/js/createOutputParameters.js
docs/ui/src/js/createPaletteCharts.js
docs/ui/src/js/createPanelReportTable.js
docs/ui/src/js/createRGBchannelChart.js
docs/ui/src/js/createRatioChart.js
docs/ui/src/js/createSVGgradient.js
docs/ui/src/js/createSVGswatches.js
docs/ui/src/js/createSVGuiKit.js
docs/ui/src/js/createSamples.js
docs/ui/src/js/createTable.js
docs/ui/src/js/createXML.js
docs/ui/src/js/cvdColors.js
docs/ui/src/js/d3.js
docs/ui/src/js/demos/demo_chord.js
docs/ui/src/js/demos/demo_choropleth.js
docs/ui/src/js/demos/demo_density.js
docs/ui/src/js/demos/demo_donut.js
docs/ui/src/js/demos/demo_heatmap.js
docs/ui/src/js/demos/demo_hexbin.js
docs/ui/src/js/demos/demo_scatter.js
docs/ui/src/js/demos/demo_stackedArea.js
docs/ui/src/js/demos/demo_violinJitter.js
docs/ui/src/js/getThemeData.js
docs/ui/src/js/initialDivergingScale.js
docs/ui/src/js/initialQualitativeScale.js
docs/ui/src/js/initialSequentialScale.js
docs/ui/src/js/initialTheme.js
docs/ui/src/js/keyColors.js
docs/ui/src/js/pageLoader.js
docs/ui/src/js/params.js
docs/ui/src/js/polarColorPath.js
docs/ui/src/js/popover.js
docs/ui/src/js/predefinedColorNames.js
docs/ui/src/js/ramps.js
docs/ui/src/js/ratios.js
docs/ui/src/js/scaleKeyColors.js
docs/ui/src/js/sliderInput.js
docs/ui/src/js/sortColorScales.js
docs/ui/src/js/tabs.js
docs/ui/src/js/themeTitle.js
docs/ui/src/js/themeUpdate.js
docs/ui/src/js/toast.js
docs/ui/src/js/toggleSwatchAttributes.js
docs/ui/src/js/tooltip.js
docs/ui/src/js/utils.js
docs/ui/src/scales.html
docs/ui/src/scales.js
docs/ui/src/scss/colorinputs.scss
docs/ui/src/scss/components/articleCard.scss
docs/ui/src/scss/components/colorPicker.scss
docs/ui/src/scss/components/colorSlider.scss
docs/ui/src/scss/components/dialog.scss
docs/ui/src/scss/components/header.scss
docs/ui/src/scss/components/outputSwatches.scss
docs/ui/src/scss/components/sections.scss
docs/ui/src/scss/components/selectBox.scss
docs/ui/src/scss/components/textfield.scss
docs/ui/src/scss/converter.scss
docs/ui/src/scss/style.scss
docs/ui/src/theme.html
docs/ui/src/theme.js
docs/ui/src/tools.html
docs/ui/src/tools.js
docs/ui/src/views/GithubLink.html
docs/ui/src/views/LogoLink.html
docs/ui/src/views/dialog_bulkColorConverter.html
docs/ui/src/views/dialog_importURL.html
docs/ui/src/views/footer.html
docs/ui/src/views/header.html
docs/ui/src/views/header_scales.html
docs/ui/src/views/header_tools.html
docs/ui/src/views/home_articles.html
docs/ui/src/views/home_contribute.html
docs/ui/src/views/home_gettingStarted.html
docs/ui/src/views/home_learn.html
docs/ui/src/views/home_main.html
docs/ui/src/views/home_sidenav.html
docs/ui/src/views/tab_colorCompare.html
docs/ui/src/views/tab_colorConverter.html
docs/ui/src/views/tab_swatches.html
docs/ui/src/views/tab_theme.html
docs/ui/src/views/toast_cvdPreview.html
package.json
packages/contrast-colors/.eslintrc.json
packages/contrast-colors/README.md
packages/contrast-colors/index.d.ts
packages/contrast-colors/index.js
packages/contrast-colors/lib/backgroundcolor.js
packages/contrast-colors/lib/chroma-plus.js
packages/contrast-colors/lib/color.js
packages/contrast-colors/lib/curve.js
packages/contrast-colors/lib/theme.js
packages/contrast-colors/lib/utils.js
packages/contrast-colors/moon.yml
packages/contrast-colors/package.json
packages/contrast-colors/test/backgroundColor.test.js
packages/contrast-colors/test/colorSetters.test.js
packages/contrast-colors/test/contrast.test.js
packages/contrast-colors/test/convertColorValue.test.js
packages/contrast-colors/test/createScale.test.js
packages/contrast-colors/test/minPositive.test.js
packages/contrast-colors/test/ratioName.test.js
packages/contrast-colors/test/searchColors.test.js
packages/contrast-colors/test/theme.test.js
packages/contrast-colors/test/themeSetters.test.js
pnpm-lock.yaml
```

</div></details>

**Comment by GarthDB** [comment]
It's close to ready I'm just fixing the tasks to not all run on CI.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  ci:
+    name: CI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: moonrepo/setup-moon-action@v1
+      - run: moon ci
```
[the remark this answers is not in the document]
Comment (GarthDB) [reply .github/workflows/ci.yml:12]: If I move the tasks to moon I can define which ones are local only.

**Pull request merged**